lib/client.js

```diff
@@ -0,0 +1,60 @@
+'use strict';
+
+const net = require('net');
+const { parse } = require('url');
+const { join } = require('path');
+const EventEmitter = require('events');
+const Connection = require('./connection');
+const fs = require('./fs-interface');
+
+const DEFAULT_TARGET_DIR = process.cwd();
+const onMessage = Symbol('onMessage');
+
+class Client extends EventEmitter {
+  constructor() {
+    super();
+
+    const socket = new net.Socket();
+    this.connection = new Connection(socket);
+
+    this.connection.transport.on('connect', () => this.emit('connect'));
+    this.connection.on('error', err => this.emit('error', err));
+    this.connection.on('close', () => this.emit('close'));
+    this.connection.on('message', this[onMessage].bind(this));
+  }
+
+  connect(url) {
```

Comment:
better create and connect `net.Socket` right here and then wrap it into `Connection` and emit `connect`